### ISSUE TITLE
Locale remapping doesn't work for gui/theme/custom_font

### ISSUE THREAD

**Issue by mysticfall** [godotengine/godot#17640]
**Godot version:**
`master` / dad47d8876de45132226985dbbd079e060f6facc

**OS/device including version:**
_Manjaro Linux 17.1_

**Issue description:**
I'm trying to use a different font per each locale using the resource remapping feature. But it seems I can't make it work whether I assign it to `Default Font` property of a UI theme, or `Custom Font` property in the project settings.

I created a `DynamicFont` resource using an English font, then duplicated it to create one for Korean locale. When I manually assign the Korean version to font related properties it works as intended but if I use the original resource, all Korean texts show up as blank space, apparently because the remapping setting is ignored.

Here's the relevant part in my `project.godot` file:
```ini
[gui]
theme/custom_font="res://Data/Fonts/UI.tres"

[locale]
locale_filter=[ 1, [ "en", "ko" ] ]
translations=PoolStringArray( "res://Data/I18n/Messages.en.translation", "res://Data/I18n/Messages.ko.translation" )
translation_remaps={
"res://Data/Fonts/UI.tres": PoolStringArray( "res://Data/Fonts/UI.ko.tres:ko" )
}
```

**Comment by reduz** [comment]
Btw, keep in mind you can use fallbacks in DynamicFont to add other charsets, so you save having to do this.

**Comment by reduz** [comment]
This is not really a bug but should probably be added as documentation (how to do locale remapping using fallbacks).

**Comment by mysticfall** [comment]
But shouldn't resource remapping work for fonts too, as long as they are resources also? I think it'd be cleaner and more consistent solution to the font problem if it works that way.

Sometimes, fonts assigned for different locales might be able to render some common characters (i.e. numbers) in which case font resolving may work differently than what the author intended.

And when there are many different locales to support, it might be more efficient to just use the font resource that matches the current setting, rather than having to test for a long list of fallback fonts per each UI element.

**Comment by rambda** [comment]
> 
> 
> Btw, keep in mind you can use fallbacks in DynamicFont to add other charsets, so you save having to do this.


For localizing into CJK language, there are four approaches that I can think of:

1. English font as primary font, CJK fonts as fallbacks. 
![图片](https://user-images.githubusercontent.com/23667522/88545908-35641780-d04e-11ea-8ac9-e63b736bc1b1.png)
Quotes are in English style, which is unwanted. I think there are other notations conflicts among languages, besides `“”‘’`.

2. Use CJK fonts only for both CJK languages and English. 
![图片](https://user-images.githubusercontent.com/23667522/88545818-149bc200-d04e-11ea-8227-08ced49ea484.png)
Quotes are in Chinese style, which is expected.
But English is displayed in the style of this CJK font. It's OK but not the style I want.

3. Suppose locale remapping for DynamicFont works. I can then map English font to en, map simplified chinese font only to zh_CN and so on... It's better this way I think.

4. Besides, I can write codes that modify every DynamicFont on language changes. It shouldn't be too hard. But still, it will be much more convenient if we can remap fonts.

Also, I think, for same amount of information, it's long in some languages, short in other languages. It may be necessary to map fonts of different sizes to locals, so a long word can fit into a button, a long paragraph can fit into a window and such... Font fallback cannot do that.

**Comment by vitorbalbio** [comment]
I opened a Proposal for it and it was closed for no reason.
https://github.com/godotengine/godot-proposals/issues/1799

**Comment by vitorbalbio** [comment]
This is what happened with my proposal about it @rambda: Ignored and Closed for no reason.
https://github.com/godotengine/godot-proposals/issues/1799
I wish better lucky for you here.

**Comment by akien-mga** [comment]
For the reference, DynamicFonts remapping *does* work. See https://github.com/godotengine/godot-proposals/issues/1799#issuecomment-771645378 and the included test project.

What doesn't work, which is a bug, is that this remapping is not handled by the `gui/theme/custom_font` project setting.

**Comment by akien-mga** [comment]
The problem is that in `main.cpp`, `register_scene_types()` (which handles loading the font for `gui/theme/custom_font`) is called before `ResourceLoader::load_translation_remaps()`. So the remapping doesn't exist yet when the custom font is loaded.

One can try to move things around but the order of those things is fairly critical, as stuff done after `register_scene_types()` likely depends on scene types being registered (like for example all Resource types).

**Comment by akien-mga** [comment]
This should fix the issue: #45662 (and also handle remapping for `gui/theme/custom` which would likely fail until now in the same way, if you want a whole different theme for some locales).

**Comment by vitorbalbio** [comment]
> For the reference, DynamicFonts remapping does work. See godotengine/godot-proposals#1799 (comment) and the included test project.

Should i have said ".ttf" files remap instead of "Dynamic Font". My mistake. It's not possible anyway.

> Your proposal was neither ignored (you had a length discussion about it with a contributor) nor closed for no reason (the reason for closing was justified to you - you can of course disagree with that reason, but that's still a reason). So please refrain from making such comment, this is not the place to express your frustration.
 
 My apologizes about my frustration. Will not insist on it. What was said and done is there to anyone to check. Let's move on.
